webots_ros2_tests/test/test_system_rosbot.py

```diff
@@ -0,0 +1,99 @@
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Test the `webots_ros2_husarion` package."""
+
+# Launch the test locally: launch_test src/webots_ros2/webots_ros2_tests/test/test_system_rosbot.py
+
+import os
+import pytest
+import rclpy
+from nav_msgs.msg import Odometry
+from launch import LaunchDescription
+from launch.launch_description_sources import PythonLaunchDescriptionSource
+import launch_testing.actions
+from launch.actions import IncludeLaunchDescription
+from geometry_msgs.msg import Twist
+from sensor_msgs.msg import LaserScan
+from ament_index_python.packages import get_package_share_directory
+from webots_ros2_tests.utils import TestWebots, initialize_webots_test
+
+
+@pytest.mark.rostest
+def generate_test_description():
+    os.environ["USERNAME"] = "root"
+    initialize_webots_test()
+    # If ROS_DISTRO is rolling, skip the test as some required packages are missing (cf. ci_after_init.bash)
+    if 'ROS_DISTRO' in os.environ and os.environ['ROS_DISTRO'] == 'rolling':
+        pytest.skip('ROS_DISTRO is rolling, skipping this test')
+
+    rosbot_webots = IncludeLaunchDescription(
+        PythonLaunchDescriptionSource(
+            os.path.join(get_package_share_directory('webots_ros2_husarion'), 'launch', 'rosbot_launch.py'),
+        )
+    )
+
+    return LaunchDescription([
+        rosbot_webots,
+        launch_testing.actions.ReadyToTest(),
+    ])
+
+
+class TestROSbot(TestWebots):
+    @classmethod
+    def setUpClass(cls):
+        rclpy.init()
+
+    @classmethod
+    def tearDownClass(cls):
+        rclpy.shutdown()
+
+    def setUp(self):
+        self.__node = rclpy.create_node('driver_tester')
+
+    def testMovement(self):
+        publisher = self.__node.create_publisher(Twist, '/cmd_vel', 1)
+
+        def on_position_message_received(message):
+            twist_message = Twist()
+            twist_message.linear.x = 0.5
+            twist_message.angular.z = 0.3
+            publisher.publish(twist_message)
+
+            # ROSbot should move in an arc to check the sensor fusion
+            if message.pose.pose.position.x > 0.5 and message.pose.pose.orientation.w < 0.9:
+                return True
+            return False
+
+        self.wait_for_messages(self.__node, Odometry, '/odometry/filtered', condition=on_position_message_received)
+
+    def testScan(self):
+        def on_scan_message_received(message):
+            # There should be at least 1 range bigger than 0 and some = 0
+            number_of_inf = 0
+            number_of_non_zeroes = 0
+            for value in message.ranges:
+                if value == float('inf'):
```

Comment:
Concerning the Scan test, I believe your implementation is correct. The E-puck one, on the other side, isn't. The e-puck node is publishing on the `/scan` topic the values of DistanceSensors (and not a Lidar). The node should be updated to use `inf` as `OUT_OF_RANGE` value. I will open an issue.